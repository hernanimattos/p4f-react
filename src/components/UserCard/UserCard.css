.user-card {
  /* box-shadow: 1px 2px 3px 1px rgba(0, 0, 0, 0.2); */
  padding: 0.5rem;
  cursor: pointer;
  margin: 0.1rem;
  transition: all 0.2s ease-in-out;
  background-color: #fff;
}
.user-card.default {
  box-shadow: 1px 2px 3px 1px rgba(0, 0, 0, 0.2);
}
.user-card.default:hover {
  transform: scale(1.05);
  background-color: rgb(26, 117, 255);
  color: #fff;
}
