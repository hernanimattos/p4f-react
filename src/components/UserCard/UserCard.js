import React, { Fragment } from 'react';
import PropTypes from 'prop-types';
import './UserCard.css';

const UserCard = (props) => {
  const { name, email, company, primary, address } = props || {};
  const { street, zipcode, suite, city } = address || {};

  return (
    <div className={!primary ? 'user-card default' : 'user-card '}>
      {!primary ? (
        <Fragment>
          <h2>{name}</h2>
          <p>{email}</p>
          <p>{company}</p>
        </Fragment>
      ) : (
        <Fragment>
          <h2>{name}</h2>
          <h3>Endere√ßo</h3>
          <p>
            {street} - {suite}
          </p>
          <p>
            {city} - CEP: {zipcode}
          </p>
        </Fragment>
      )}
    </div>
  );
};

UserCard.propTypes = {
  name: PropTypes.string,
  email: PropTypes.string,
  street: PropTypes.string,
  suite: PropTypes.string,
  city: PropTypes.string,
  zipcode: PropTypes.string,
};

export default UserCard;
