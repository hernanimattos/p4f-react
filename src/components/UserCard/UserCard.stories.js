import React from 'react';
import UserCard from './UserCard';
import { withKnobs } from '@storybook/addon-knobs';
// import { action } from '@storybook/addon-actions';

export default {
  title: 'UserCard',
  component: UserCard,
  decorators: [withKnobs],
};

export const primary = () => (
  <UserCard
    name="Eu mesmo"
    email="eumesmo@eumesmo.nois"
    company="company e sempany"
  />
);

export const notprimary = () => {
  const address = {
    street: 'figth',
    suite: '3b',
    city: 'Gotham',
    zipcode: '000000',
  };

  return <UserCard name="Eu mesmo" address={address} primary />;
};
