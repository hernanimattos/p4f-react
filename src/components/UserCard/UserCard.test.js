import React from 'react';
import { render, waitForDomChange } from '@testing-library/react';
import UserCard from './UserCard';

test('Snap', () => {
  const UserCardCP = UserCard.toString();
  expect(UserCardCP).toMatchSnapshot();
});

test('PostCard class', () => {
  const { container } = render(<UserCard />);
  expect(container.firstChild).toHaveClass('user-card');
});

test('PostCard class', () => {
  const props = {
    address: {
      street: 'aaaaa',
    },
    name: 'nome',
    email: 'email',
  };
  const { getByText, container } = render(<UserCard primary {...props} />);
  expect(container.firstChild).toHaveClass('user-card default');
  expect(getByText('nome')).toBeInTheDocument();
  expect(getByText('Endere√ßo')).toBeInTheDocument();
});
