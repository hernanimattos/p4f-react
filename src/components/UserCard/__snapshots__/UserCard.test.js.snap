// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Snap 1`] = `
"props => {
  const {
    name,
    email,
    company,
    primary,
    address
  } = props || {};
  const {
    street,
    zypcode,
    suite,
    city
  } = address || {};
  return /*#__PURE__*/_react.default.createElement(\\"div\\", {
    className: primary ? 'user-card default' : 'user-card ',
    __self: void 0,
    __source: {
      fileName: _jsxFileName,
      lineNumber: 9,
      columnNumber: 5
    }
  }, !primary ? /*#__PURE__*/_react.default.createElement(_react.Fragment, {
    __self: void 0,
    __source: {
      fileName: _jsxFileName,
      lineNumber: 11,
      columnNumber: 9
    }
  }, /*#__PURE__*/_react.default.createElement(\\"h2\\", {
    __self: void 0,
    __source: {
      fileName: _jsxFileName,
      lineNumber: 12,
      columnNumber: 11
    }
  }, name), /*#__PURE__*/_react.default.createElement(\\"p\\", {
    __self: void 0,
    __source: {
      fileName: _jsxFileName,
      lineNumber: 13,
      columnNumber: 11
    }
  }, email), /*#__PURE__*/_react.default.createElement(\\"p\\", {
    __self: void 0,
    __source: {
      fileName: _jsxFileName,
      lineNumber: 14,
      columnNumber: 11
    }
  }, company)) : /*#__PURE__*/_react.default.createElement(_react.Fragment, {
    __self: void 0,
    __source: {
      fileName: _jsxFileName,
      lineNumber: 17,
      columnNumber: 9
    }
  }, /*#__PURE__*/_react.default.createElement(\\"h2\\", {
    __self: void 0,
    __source: {
      fileName: _jsxFileName,
      lineNumber: 18,
      columnNumber: 11
    }
  }, name), /*#__PURE__*/_react.default.createElement(\\"h3\\", {
    __self: void 0,
    __source: {
      fileName: _jsxFileName,
      lineNumber: 19,
      columnNumber: 11
    }
  }, \\"Endere\\\\xE7o\\"), /*#__PURE__*/_react.default.createElement(\\"p\\", {
    __self: void 0,
    __source: {
      fileName: _jsxFileName,
      lineNumber: 20,
      columnNumber: 11
    }
  }, street, \\" - \\", suite), /*#__PURE__*/_react.default.createElement(\\"p\\", {
    __self: void 0,
    __source: {
      fileName: _jsxFileName,
      lineNumber: 23,
      columnNumber: 11
    }
  }, city, \\" - CEP: \\", zypcode)));
}"
`;
